File: client-data/src/test/java/com/jraska/github/client/FakeConfig.java
package com.jraska.github.client;

import com.jraska.github.client.common.Maps;

import java.util.Collections;
import java.util.Map;

public final class FakeConfig implements Config {
  private final Map<String, Object> values;

  public static FakeConfig create() {
    return create(Collections.emptyMap());
  }

  public static FakeConfig create(Map<String, Object> values) {
    return new FakeConfig(Collections.unmodifiableMap(values));
  }

  public static FakeConfig create(String key, Object value) {
    return create(Maps.newHashMap(key, value));
  }

  private FakeConfig(Map<String, Object> values) {
    this.values = values;
  }

  @Override public void triggerRefresh() {
    // do nothing
  }

  @Override public boolean getBoolean(String key) {
    Object value = values.get(key);
    return value != null && (boolean) value;
  }

  @Override public String getString(String key) {
    Object value = values.get(key);
    if (value != null) {
      return (String) value;
    }

    return "";
  }

  @Override public long getLong(String key) {
    Object value = values.get(key);
    if (value != null) {
      return (long) value;
    }

    return 0;
  }
}

File: client-data/src/test/java/com/jraska/github/client/FakeConfig.kt
package com.jraska.github.client

import java.util.*


class FakeConfig private constructor(private val values: Map<String, Any>) : Config {

  override fun triggerRefresh() {
    // do nothing
  }

  override fun getBoolean(key: String): Boolean {
    return values[key] as Boolean
  }

  override fun getString(key: String): String {
    return values[key] as String
  }

  override fun getLong(key: String): Long {
    return values[key] as Long
  }

  companion object {
    fun create(values: Map<String, Any> = emptyMap()): FakeConfig {
      return FakeConfig(Collections.unmodifiableMap(values))
    }
  }
}
